### ISSUE TITLE
Style/MultilineWhenThen breaks some single line case statements

### ISSUE THREAD

**Issue by tas50** [rubocop-hq/rubocop#7635]
# Expected behavior

Style/MultilineWhenThen should detect multiline when statements with a then, but should not detect strange compact when/then statements.

## Actual behavior

Given a compact case statement like this rubocop alerts Style/MultilineWhenThen:

```ruby
def global_shell_init(action = nil)
  matcher = /^eval "\$\(chef shell-init bash\)"$/
  line = 'eval "$(chef shell-init bash)"'
  @global_shell_init ||= Chef::Util::FileEdit.new(bashrc_file)
  return @global_shell_init unless new_resource.global_shell_init
  case action
  when :create then @global_shell_init.insert_line_if_no_match(matcher,
                                                               line)
  when :delete then @global_shell_init.search_file_delete_line(matcher)
  end
  @global_shell_init
end
```

```
chef-dk/libraries/provider_chef_dk.rb:128:22: C: [Corrected] Style/MultilineWhenThen: Do not use then for multiline when statement.
        when :create then @global_shell_init.insert_line_if_no_match(matcher,
                     ^^^^
```

This will autocorrect to:

```ruby
def global_shell_init(action = nil)
  matcher = /^eval "\$\(chef shell-init bash\)"$/
  line = 'eval "$(chef shell-init bash)"'
  @global_shell_init ||= Chef::Util::FileEdit.new(bashrc_file)
  return @global_shell_init unless new_resource.global_shell_init
  case action
  when :create @global_shell_init.insert_line_if_no_match(matcher,
                                                               line)
  when :delete then @global_shell_init.search_file_delete_line(matcher)
  end
  @global_shell_init
end
```

## RuboCop version

```
0.79.0
```

**Comment by koic** [comment]
The given code and the auto-corrected code are the same. Can you show me a reproducible given code?

**Comment by tas50** [comment]
🤦‍♂ Let me find the correct example code

**Comment by tas50** [comment]
Alright I've updated to show that RuboCop just nukes the first `then` and breaks the `when`.

**Comment by tagliala** [comment]
Hi, this is still an issue with RuboCop 0.82.0

I have a working reduced test case

### Original file
```rb
# frozen_string_literal: true

def bar
  []
end

def foo(arg)
  case arg
  when 'one'   then [1]
  when 'two'   then bar ||
                    [3]
  when 'three' then [4]
  end
end

puts foo('one')
```

### Run Original File
```
$ ruby issue_7635.rb 
1
```

### RuboCop output

```
$ rubocop 
Inspecting 1 file
C

Offenses:

issue_7635.rb:10:16: C: Style/MultilineWhenThen: Do not use then for multiline when statement.
  when 'two'   then bar ||
               ^^^^
issue_7635.rb:11:21: C: Layout/MultilineOperationIndentation: Use 2 (not 18) spaces for indenting an expression spanning multiple lines.
                    [3]
                    ^^^

1 file inspected, 2 offenses detected
```


### RuboCop autofix (safe)

```
$ rubocop -a --safe
Inspecting 1 file
E

Offenses:

issue_7635.rb:10:14: E: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  when 'two' bar ||
             ^^^
issue_7635.rb:10:16: C: [Corrected] Style/MultilineWhenThen: Do not use then for multiline when statement.
  when 'two'   then bar ||
               ^^^^
issue_7635.rb:11:21: C: [Corrected] Layout/MultilineOperationIndentation: Use 2 (not 18) spaces for indenting an expression spanning multiple lines.
                    [3]
                    ^^^

1 file inspected, 3 offenses detected, 2 offenses corrected
```

### Fixed file

```rb
# frozen_string_literal: true

def bar
  []
end

def foo(arg)
  case arg
  when 'one'   then [1]
  when 'two' bar ||
    [3]
  when 'three' then [4]
  end
end

puts foo('one')
```

### Run fixed file
```
$ ruby issue_7635.rb 
issue_7635.rb:10: syntax error, unexpected local variable or method, expecting `then' or ',' or ';' or '\n'
  when 'two' bar ||
```
